### PR TITLE
Invoker refactor 1

### PR DESCRIPTION
**Pull request by janezpodhostnik**
## Goal of this PR

Refactor invoker:

- some renaming
- remove error text duplication
- remove redundant check ValidateHeightIndexed when getting account keys
- skip creating a dictionary of account keys then getting one key from it
- move some code closer together
- execute script accepts byte arrays instead of cadence values  so we don't have to decode+encode+decode them before they are used.

### Misc

- [ ] PR title will be clear as part of the changelog
- [ ] PR is against the correct branch
- [ ] PR is labelled appropriately
- [ ] PR is linked to an issue

### CODE REVIEW

File: service/invoker/invoker.go [hunk 1]
```diff
@@ -195,3 +162,40 @@ func (i *Invoker) ByHeightFrom(height uint64, header *flow.Header) (*flow.Header
...
+		readRegister(i.index, i.cache, height))
+}
+
+func readRegister(
+	index archive.Reader,
+	cache Cache,
+	height uint64,
+) func(flow.RegisterID) (flow.RegisterValue, error) {
+	return func(regID flow.RegisterID) (flow.RegisterValue, error) {
+		cacheKey := fmt.Sprintf("%d/%s", height, regID)
```
Comment (zhangchiqing) [inline service/invoker/invoker.go:182]: we should [reuse this function and make it public](https://github.com/onflow/flow-archive/blob/master/service/storage2/payload/lookup.go#L18)

**Review by zhangchiqing — APPROVED** [review]

File: service/invoker/invoker.go (hunk 1 above)
Comment (janezpodhostnik) [reply service/invoker/invoker.go:182]: not sure I understand. that function looks like it does something else...

**Review by koko1123 — APPROVED** [review]

File: service/invoker/invoker.go (hunk 1 above)
Comment (zhangchiqing) [reply service/invoker/invoker.go:182]: nvm, I think we can use a different key for the caching as long as its unique

File: cmd/flow-archive-client/main.go [hunk 2]
```diff
@@ -95,15 +94,16 @@ func run() int {
...
 	if flagParams != "" {
 		params := strings.Split(flagParams, ",")
 		for _, param := range params {
-			arg, err := convert.ParseCadenceArgument(param)
+			carg, err := convert.ParseCadenceArgument(param)
 			if err != nil {
 				log.Error().Err(err).Msg("invalid Cadence value")
 				return failure
 			}
+			arg, err := json.Encode(carg)
```
Comment (koko1123) [inline cmd/flow-archive-client/main.go:106]:
```suggestion
			arg, _ := json.Encode(carg)
```

Comment (koko1123) [reply cmd/flow-archive-client/main.go:106]: not advocating for ignoring the error

Comment (janezpodhostnik) [reply cmd/flow-archive-client/main.go:106]: Yeah, I overlooked that, but it is in https://github.com/onflow/flow-archive/pull/136 that I will merge into this one.

**Pull request merged**